plaid/auth.go

```diff
@@ -1,207 +1,58 @@
 package plaid
 
 import (
-	"bytes"
 	"encoding/json"
+	"errors"
 )
 
-// AuthAddUser (POST /auth) submits a set of user credentials to add an Auth user.
-//
-// See https://plaid.com/docs/api/#add-auth-user.
-func (c *Client) AuthAddUser(username, password, pin, institutionType string,
-	options *AuthOptions) (postRes *postResponse, mfaRes *mfaResponse, err error) {
-
-	jsonText, err := json.Marshal(authJson{
-		ClientID: c.clientID,
-		Secret:   c.secret,
-		Type:     institutionType,
-		Username: username,
-		Password: password,
-		PIN:      pin,
-		Options:  options,
-	})
-	if err != nil {
-		return nil, nil, err
-	}
-	return c.postAndUnmarshal("/auth", bytes.NewReader(jsonText))
+type authGetRequestOptions struct {
+	AccountIDs []string `json:"account_ids"`
 }
 
-// AuthStepSendMethod (POST /auth/step) specifies a particular send method for MFA,
-// e.g. `{"mask":"xxx-xxx-5309"}`.
-//
-// See https://plaid.com/docs/api/#auth-mfa.
-func (c *Client) AuthStepSendMethod(accessToken, key, value string) (postRes *postResponse,
-	mfaRes *mfaResponse, err error) {
-
-	sendMethod := map[string]string{key: value}
-	jsonText, err := json.Marshal(authStepSendMethodJson{
-		ClientID:    c.clientID,
-		Secret:      c.secret,
-		AccessToken: accessToken,
-		Options:     authStepOptions{sendMethod},
-	})
-	if err != nil {
-		return nil, nil, err
-	}
-	return c.postAndUnmarshal("/auth/step", bytes.NewReader(jsonText))
+type authRequestJson struct {
+	ClientID    string                `json:"client_id"`
+	Secret      string                `json:"secret"`
+	AccessToken string                `json:"access_token"`
+	Options     authGetRequestOptions `json:"options"`
```

Comment:
@evanlimanto - can we replace `omitempty` for all other `Options` in the Go SDK? Since there are by default optional and should not be included in the API request if omitted

Reply:
done!